crates/codegen/runtime/cargo/crate/src/ir/builder.rs.jinja2

```diff
@@ -181,32 +170,49 @@
       self.index < self.children.len() && self.children[self.index].label == label
     }
 
-    fn accept_label(&mut self, label: EdgeLabel) -> Result<&Node> {
+    fn accept_label(&mut self, label: EdgeLabel) -> Option<&Node> {
       if self.index >= self.children.len() {
-        return Err(format!("Expected more sibling nodes, looking for label {label:?}"));
+        // we may end up in this case if the parser generates an incomplete CST
+        // node because of a parser error
+        return None;
       }
-      if self.children[self.index].label == label {
-        let node = &self.children[self.index].node;
+      if !self.children[self.index].is_valid() {
+        // attempt to recover from an unrecognized/missing node
         loop {
           self.index += 1;
-          if self.index >= self.children.len() || !self.children[self.index].is_trivia() {
+          if self.index >= self.children.len() {
+            return None;
+          }
+          if !self.children[self.index].is_trivia() {
```

Comment:
I wonder if we can simply skip over invalid nodes, and still look for the label after it?
This would protect us against an unrecognized character, where the parser resumed normally afterwards.

```suggestion
          if !self.children[self.index].is_trivia() && self.children[self.index].is_valid() {
```

Reply:
Ditto.